crypto/src/main/java/net/corda/v5/crypto/DigitalSignature.java

```diff
@@ -16,19 +14,19 @@ public DigitalSignature(@NotNull byte[] bytes) {
     }
 
     /**
-     * A digital signature that identifies who the public key is owned by.
+     * A digital signature that identifies who is the owner of the signing key used to create this signature.
      */
-    public static class WithKey extends DigitalSignature {
+    public static class WithKeyId extends DigitalSignature {
 
         /**
-         * Construct WithKey
+         * Creates a new {@code WithKeyId} using the specified key id and the signature ({@code bytes}).
          *
          * @param by      The public key of the corresponding private key used to sign the data (as if an instance
          *                of the {@link CompositeKey} is passed to the sign operation it may contain keys which are not actually owned by
          *                the member).
```

Comment:
```suggestion
         * @param by      The id of the key pair used to sign the data (if this resolves to a
         *                {@link CompositeKey}, not all keys might be owned by the member).
```

This needs to be updated, as it doesn't take a key anymore. Also, the wording around composite keys is a bit confusing.

Reply:
I don't think that's very relevant to be here. Maybe more relevant to say that if composite key used to sign it will sign with the firstly found key leaf.